onnxruntime/core/graph/graph_utils.cc

```diff
@@ -42,6 +42,17 @@ bool IsSupportedOptypeVersionAndDomain(const Node& node,
   return true;
 }
 
+bool IsSupportedProvider(const Node& node,
+                         const std::vector<std::string>& compatible_providers) {
+  if (!compatible_providers.empty() && 
+      std::find(compatible_providers.cbegin(), compatible_providers.cend(), 
+          node.GetExecutionProviderType()) == compatible_providers.cend()) {
+    return false;
+  }
+  
+  return true;
+}
```

Comment:
Are we treating an empty list as meaning match all? If so, the description for IsSupportedProvider should call that out.